### PR TITLE
Remove decoration when href is missing

### PR DESCRIPTION
**Pull request by EA-YOUHOU**
A tag must not have decoration when href is missing.

### CODE REVIEW

**Comment by EA-YOUHOU** [comment]
I've some trouble to understand why this PR fail in the format test...

**Comment by daohoangson** [comment]
Looks like there are some new lint rules so they are flagging other files. Let me deal with those first, then you can rebase your PR later.

Btw, an empty href should be treated in the same way as no href I think?

**Comment by EA-YOUHOU** [comment]
A tag with empty href must appear like filled href.
A tag with missing href must not show any decoration / underline.

**Comment by daohoangson** [comment]
> A tag with empty href should appear like filled href.
> A tag with missing href should not show any decoration / underline.

You are right, please ignore my earlier comment 🙈

File: packages/core/lib/src/core_widget_factory.dart [hunk 1]
```diff
@@ -639,13 +639,15 @@ class WidgetFactory {
         _tagA ??= TagA(this).buildOp;
         meta.register(_tagA!);
```
Comment (daohoangson) [inline packages/core/lib/src/core_widget_factory.dart:640]: Maybe we should check for `href` before registering the build op?

**Comment by daohoangson** [comment]
Also: please add a few test to make sure it works for cases as you listed here #676

```html
<a>test</a> no underline and no clickable effect
<a href>test</a> underline and clickable effect
<a href="">test</a> underline and clickable effect
<a href="test">test</a> underline and clickable effect
```

File: packages/core/lib/src/core_widget_factory.dart (hunk 1 above)
Comment (EA-YOUHOU) [reply packages/core/lib/src/core_widget_factory.dart:640]: I didn't know, i didn't spend enough time on the code to suggest a better solution :/

**Comment by EA-YOUHOU** [comment]
@daohoangson test added :)

**Pull request merged**